platform/src/test/python/dlpx/virtualization/test_migration_helper.py

```diff
@@ -0,0 +1,359 @@
+#
+# Copyright (c) 2019 by Delphix. All rights reserved.
+#
+
+import pytest
+from dlpx.virtualization.platform import migration_helper as m
+from dlpx.virtualization.platform.exceptions import (
+    MigrationIdAlreadyUsedError, MigrationIdIncorrectFormatError,
+    MigrationIdIncorrectTypeError)
+
+
+class TestPlatformUpgradeMigrations:
+    @staticmethod
+    @pytest.fixture
+    def platform_migrations():
+        yield m.PlatformUpgradeMigrations()
+
+    @staticmethod
+    @pytest.mark.parametrize('method_name', [
+        'add_repository', 'add_source_config', 'add_linked_source',
+        'add_virtual_source', 'add_snapshot'
+    ])
+    @pytest.mark.parametrize('migration_id,expected_std_id',
+                             [('5.3.2.1', '5.3.2.1'), ('1000', '1000'),
+                              ('50.0.0', '50'), ('50.0.0000.1', '50.0.0.1'),
+                              ('2019.10.04', '2019.10.4')])
```

Comment:
I notice that you have the same list of functions/migration_ids in multiple tests. Maybe just make a single list and pass it into each decorator?
```
method_names = [
    'add_repository', 'add_source_config', 'add_linked_source',
    'add_virtual_source', 'add_snapshot'
]

@pytest.mark.parametrize('method_name', method_names)
...
```
And same with the migration_ids?